quickwit/quickwit-storage/src/object_storage/azure_compatible_storage.rs

```diff
@@ -302,9 +303,14 @@ impl Storage for AzureBlobStorage {
         let mut output_stream = self.container_client.blob_client(name).get().into_stream();
 
         let mut dest_file = File::create(output_path).await?;
-        while let Some(result) = output_stream.next().await {
-            let chunk = result.map_err(AzureErrorWrapper::from)?.data;
-            dest_file.write_all(&chunk).await?;
+        while let Some(chunk_result) = output_stream.next().await {
+            let chunk_response = chunk_result.map_err(AzureErrorWrapper::from)?;
+            let chuck_data = chunk_response
```

Comment:
Do we have to collect here? Can we put a buf writer in front of the file and stream the data into the file directly? Or at least reuse the same vec between chunks?